### PR TITLE
Integration tests setup

### PR DESCRIPTION
**Pull request by pbabic-redhat**
## What

* Adds a job to the existing GH actions workflow to run integration tests
* Adds integration testing directory structure and a test image and package (from PKO)
* Adds integration tests for `dev/build.go`

## JIRA

[MTSRE-767](https://issues.redhat.com/browse/MTSRE-767)

## Testing
* Ran locally: `$ mage test:integration`
* And in test PR in fork: [PR#3](https://github.com/pbabic-redhat/devkube/pull/3)

### CODE REVIEW

**Comment by Nanyte25** [comment]
@pbabic-redhat Just trying to test this locally for you 

```


devkube (feat-integration-tests✔)mage test:integration
go: downloading github.com/stretchr/testify v1.7.0
go: downloading k8s.io/apimachinery v0.23.1
go: downloading k8s.io/api v0.23.0
go: downloading k8s.io/apiextensions-apiserver v0.23.0
go: downloading k8s.io/client-go v0.23.0
go: downloading sigs.k8s.io/controller-runtime v0.11.0
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.2.0
go: downloading k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b
go: downloading golang.org/x/sys v0.0.0-20211029165221-6e7872819dc8
go: downloading golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
go: downloading k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65
go: downloading golang.org/x/term v0.0.0-20210615171337-6886f2dfbf5b
go: downloading golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f
go: downloading github.com/googleapis/gnostic v0.5.5
?   	github.com/mt-sre/devkube/integration/test-data/test-stub	[no test files]

```

**Comment by pbabic-redhat** [comment]
Thanks for testing it! That's strange, I cloned the repo, switched to the branch and ran the command and it worked, even after running `go clean -modcache`.

**Comment by pbabic-redhat** [comment]
@Nanyte25 is that the whole output, it ran no tests?

File: integration/test-data/test-stub/main.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	"syscall"
+)
+
+// Test Stub to use as workload stand-in in integration tests.
+func main() {
+	fmt.Fprintln(os.Stdout, "waiting for something interesting to happen...")
+
+	// block forever
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
```
Comment (eqrx) [inline integration/test-data/test-stub/main.go:16]: the `syscall` pkg is deprecated. Use `os.Interrupt` insted of `syscall.SIGINT` and `syscall.SIGTERM`

**Review by eqrx — CHANGES_REQUESTED** [review]

File: integration/test-data/test-stub/main.go (hunk 1 above)
Comment (pbabic-redhat) [reply integration/test-data/test-stub/main.go]: Thanks for the catch! Should be better now.

**Pull request merged**